### ISSUE TITLE
Refactor the MacroProcessor

### ISSUE THREAD

**Issue by SuperFola** [ArkScript-lang/Ark#242]
As suggested on our Discord server by @rakista112:
> since registerMacro/execute pass the node/macro along a pipeline, we could split them up using the Chain of Responsibility pattern.
So we may define separate handlers for Symbol macros, Condition macros, Keyword macros, Comparator macros, etc...
We can then build a pipeline and pass the Node to whoever may handle it

The idea is to make the macro processor easily extendable

**Comment by rakista112** [comment]
I'll take this

**Comment by rakista112** [comment]
So I have a WIP refactor for the pipeline. At least for the execute methods...
Ark works as usual, just with the macro executors split into classes and built into a pipeline
So I prepare a pipeline (actually just a list of Executors) with the relevant MacroProcessor methods (find_nearest_macro, registerMacro, isTruthy, evaluate, apply_to, throwMacroProcessingError, func_execute) passed to it https://github.com/rakista112/Ark/blob/feat/refactor-macroprocessor/src/Compiler/MacroExecutors/MacroExecutorPipeline.cpp

and I've split the execute method sections into separate classes that implement their own execute method.
https://github.com/rakista112/Ark/blob/feat/refactor-macroprocessor/src/Compiler/MacroExecutors/SymbolExecutor.cpp

Macroprocessor just needs to pass the executors to the pipeline and call execute to pass the Node through all of them
Also I noticed that apply_to and isTruthy only ever occured in the Conditional macro so I think we should move it to be used only in ConditionalMacro
I did however need to pass a lot of methods from MacroProcessor so we could decide if we should split these into other classes or retain it for now and focus other things like splitting up the MacroProcessor into a pipeline as well. If we're okay with heading in this pipeline direction
